ms/MeasurementSets/MeasurementSet.cc

```diff
@@ -261,27 +261,30 @@ MeasurementSet::MeasurementSet(const MeasurementSet &other)
   hasBeenDestroyed_p(False)
```

Comment:
The default constructor, which yields a Null MS, initializes `hasBeenDestroyed_p` to True. Can you initialize `hasBeenDestroyed_p` to `other.isNull()` here? Then you also don't need to modify the destructor.

Reply:
I see you now set `hasBeenDestroyed_p` to `other.hasBeenDestroyed_p` if `other` is not null. I think you can directly initialize `hasBeenDestroyed_p` to `other.hasBeenDestroyed_p` here instead, which is a littile simpler.